src/v/cluster/BUILD

```diff
@@ -86,3 +86,24 @@ redpanda_cc_library(
         "@seastar",
     ],
 )
+
+redpanda_cc_library(
+    name = "notification",
+    hdrs = [
+        "notification.h",
+    ],
+    include_prefix = "cluster",
+    visibility = ["//visibility:public"],
+    deps = [
+        "//src/v/utils:named_type",
+    ],
+)
+
+redpanda_cc_library(
+    name = "snapshot",
```

Comment:
these are used by raft, but we can't depend on the entire cluster library

Reply:
I thought STMs were decentralized at this point? Is there more to do?

Ideally we should have a DAG where raft doesn't depend on anything in cluster. We can enforce this with visibility, but that shouldn't happen in this PR

Reply:
>I thought STMs were decentralized at this point? Is there more to do?

to be fair this dependency is just the string names of the STMs. it does seem like the `persisted_stm` in raft should perhaps be pulled out? cc @mmaslankaprv

Reply:
i think `persisted_stm` should still be part of raft but ` stm_snapshot_names()` should be extracted to cluster module

Reply:
cool sgtm

Reply:
but doesn't that create a weird thing like why is there a dependency between the names of the STMs and the raft layer?

Reply:
We have dynamic registration of STMs, can we add the snapshot names in there somehow?